apps/tailwind-components/components/form/FieldInput.vue

```diff
@@ -0,0 +1,129 @@
+<script setup lang="ts">
+import type {
+  InputString,
+  InputTextArea,
+  InputRef,
+  InputPlaceHolder,
+} from "#build/components";
+import type {
+  columnId,
+  columnValue,
+  CellValueType,
+  columnValueObject,
+} from "../../../metadata-utils/src/types";
+
+type inputComponent =
+  | InstanceType<typeof InputString>
+  | InstanceType<typeof InputTextArea>
+  | InstanceType<typeof InputRef>
+  | InstanceType<typeof InputPlaceHolder>;
+
+defineProps<{
+  type: CellValueType;
+  id: columnId;
+  label: string;
+  required: boolean;
+  data: columnValue;
+  refSchemaId?: string;
+  refTableId?: string;
+  refLabel?: string;
+}>();
+
+defineEmits(["focus", "error", "update:modelValue"]);
+defineExpose({ validate });
+
+const input = ref<inputComponent>();
+
+function validate(value: columnValue) {
```

Comment:
we don't need validate anymore

Reply:
however, might be interesting case to check if central validation already worked for file. I suppose it did.